WooCommerce/Classes/ViewRelated/Notifications/ReviewDetailsViewController.swift

```diff
@@ -0,0 +1,501 @@
+        let row = detailsForRow(at: indexPath)
+        switch row {
+        case .header:
+            openProductURL()
+        default:
+            break
+        }
+    }
+}
+
+
+// MARK: - Cell Setup
+//
+private extension ReviewDetailsViewController {
+
+    /// Main Cell Setup Method
+    ///
+    func setup(cell: UITableViewCell, at row: Row) {
+        switch row {
+        case .header:
+            setupHeaderPlainCell(cell)
+        case .content:
+            setupCommentCell(cell)
+        }
+    }
+
+
+    /// Setup: Header Cell (Plain)
+    ///
+    func setupHeaderPlainCell(_ cell: UITableViewCell) {
+        guard let headerCell = cell as? NoteDetailsHeaderPlainTableViewCell else {
+            return
+        }
+
+        headerCell.leftImage = .productImage
+        headerCell.rightImage = .externalImage
+        headerCell.plainText = product?.name ?? "Unknown"
+    }
+
+
+    /// Setup: Comment Cell
+    ///
+    func setupCommentCell(_ cell: UITableViewCell) {
+        guard let commentCell = cell as? NoteDetailsCommentTableViewCell else {
+                return
+        }
+
+        // Setup: Properties
+        commentCell.titleText = productReview.reviewer
+        commentCell.detailsText = ReviewAge.from(startDate: productReview.dateCreated, toDate: Date()).description
+
+        let attributes: [NSAttributedString.Key: Any] = [.paragraphStyle: NSParagraphStyle.body,
+                                                         .font: UIFont.body,
+                                                         .foregroundColor: StyleManager.defaultTextColor]
+        commentCell.commentAttributedText = NSAttributedString(string: productReview.review.strippedHTML, attributes: attributes).trimNewlines()
+
+        commentCell.starRating = productReview.rating
+
+        let gravatar = productReview.reviewerAvatarURL
+        let gravatarURL = URL(string: gravatar)
+        commentCell.downloadGravatar(with: gravatarURL)
+
+        commentCell.isApproveEnabled  = true
+        commentCell.isTrashEnabled    = true
+        commentCell.isSpamEnabled     = true
+        commentCell.isApproveSelected = productReview.status == .approved
+
+        let reviewID = productReview.reviewID
+        let reviewSiteID = productReview.siteID
+
+        commentCell.onApprove = { [weak self] in
+            guard let self = self else {
+                return
+            }
+
+            ServiceLocator.analytics.track(.notificationReviewApprovedTapped)
+            ServiceLocator.analytics.track(.notificationReviewAction, withProperties: ["type": ProductReviewStatus.approved.rawValue])
+
+            self.moderateReview(siteID: reviewSiteID, reviewID: reviewID, doneStatus: .approved, undoStatus: .hold)
+        }
+
+        commentCell.onUnapprove = { [weak self] in
+            guard let self = self else {
+                return
+            }
+            ServiceLocator.analytics.track(.notificationReviewApprovedTapped)
```

Comment:
might we want to track a different event like `notificationReviewUnapprovedTapped`? or just use the event below with `type` property?

Reply:
That event corresponds to `review_detail_approve_button_tapped`, which in the spreadsheet is marked as a UI event, associated with the button. So, we send that event, that indicates that the button was tapped, and then another event that passes the type of status, in sequence:

```
                ServiceLocator.analytics.track(.notificationReviewApprovedTapped)
                ServiceLocator.analytics.track(.notificationReviewAction, withProperties: ["type": CommentStatus.unapproved.rawValue])
```